Framework/include/QualityControl/CommonInterface.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   CommonInterface.h
+/// \author Barthelemy von Haller
+///
+
+#ifndef QUALITYCONTROL_COMMONINTERFACE_H
+#define QUALITYCONTROL_COMMONINTERFACE_H
+
+#include <unordered_map>
+#include <Rtypes.h>
+#include <CCDB/CcdbApi.h>
+
+#include "QualityControl/QcInfoLogger.h"
+
+namespace o2::quality_control::core
+{
+
+/// \brief  Common interface for Check and Task Interfaces.
+///
+/// \author Barthelemy von Haller
+class CommonInterface
```

Comment:
I am not sure if I like the name, it does not tell us what is it *common* for. How about `UserAlgorithmInterface` or `UserCodeInterface`?